pkg/apis/extensions/install/install.go

```diff
@@ -43,11 +43,9 @@ var availableVersions = []unversioned.GroupVersion{v1beta1.SchemeGroupVersion}
 
 func init() {
 	externalVersions := []unversioned.GroupVersion{}
```

Comment:
Based on all the comments below, the high level flow here should be:

func init() {
  registered.RegisterVersion()
  if (shouldEnableVersion()) {
     enableVersion()
     registered.AddEnabledVersion()
  }
}
This should make the code easier to understand.